demos/maven/README.md

```diff
@@ -0,0 +1,93 @@
+# Configure Maven
+
+Basic configuration of [Maven](https://plugins.jenkins.io/maven-plugin/)
+
+## automated tool installation on agents
+
+```yaml
+tool:
+  maven:
+    installations:
+      - name: maven3
+        home: "/maven3"
+        properties:
+          - installSource:
+              installers:
+                - maven:
+                    id: "3.8.4"
+```
+
+## global configuration
+
+### Using Maven default files
+
+```yaml
+tool:
+  mavenGlobalConfig:
+    globalSettingsProvider: "standard"
+    settingsProvider: "standard"
+```
+
+### Using given files
+
+```yaml
+tool:
+  mavenGlobalConfig:
+    globalSettingsProvider:
+      filePath:
+        path: "/conf/maven/global-settings.xml"
+    settingsProvider:
+      filePath:
+        path: "/conf/maven/settings.xml"
+```
+
+### Using a configured config file
```

Comment:
I did not add test for this one, because it concerns the config-file-provider plugin, and is already tested